icaruscode/PMT/OpReco/ICARUSFlashAssAna_module.cc

```diff
@@ -0,0 +1,457 @@
+
+    if ( ophit->PE() < fPEOpHitThreshold ) { continue; }
+
+    const int channel_id = ophit->OpChannel(); 
+
+    sumpe_map[ channel_id ]+=ophit->PE() ;
+
+    //xyz[0] += m_pmt_x[channel_id]*ophit->PE();
+    //xyz[1] += m_pmt_y[channel_id]*ophit->PE();
+    //xyz[2] += m_pmt_z[channel_id]*ophit->PE();
+
+    m_channel_id = channel_id;
+    m_integral = ophit->Area(); // in ADC x tick
+    m_amplitude = ophit->Amplitude(); // in ADC
+    m_start_time = ophit->PeakTime();
+    m_width = ophit->Width();
+    m_abs_start_time = ophit->PeakTimeAbs();
+    m_pe = ophit->PE();
+    m_fast_to_total = ophit->FastToTotal();
+
+    ophittree->Fill();
+
+  }
+
+  m_multiplicity_left = std::accumulate( sumpe_map.begin(), sumpe_map.end(), 0,
+					 [&](int value, const std::map<int, float>::value_type& p) {
+					   return getSideByChannel(p.first)==0 ? ++value : value ;
+					 });
+
+  m_multiplicity_right =std::accumulate( sumpe_map.begin(), sumpe_map.end(), 0,
+					 [&](int value, const std::map<int, float>::value_type& p) {
+					   return getSideByChannel(p.first)==1 ? ++value : value ;
+					 });
+
+  m_sum_pe_left = std::accumulate( sumpe_map.begin(), sumpe_map.end(), 0.0, 
+				   [&](float value, const std::map<int, float>::value_type& p) {
+				     return getSideByChannel(p.first)==0 ? value+p.second : value ; 
+				   });
+  
+  m_sum_pe_right = std::accumulate( sumpe_map.begin(), sumpe_map.end(), 0.0,
+            [&](float value, const std::map<int, float>::value_type& p) {
+              return getSideByChannel(p.first)==1 ? value+p.second : value ;
+            }); 
+
+  //for( int i=0; i<3; i++ ){ xyz[i] /= (m_sum_pe_left+ m_sum_pe_right); }
+  
+}
+
+
+void opana::ICARUSFlashAssAna::endJob() {
+
+}
+
+
+void opana::ICARUSFlashAssAna::analyze(art::Event const& e) {
+
+
+  m_run = e.id().run();
+  m_event = e.id().event();
+  m_timestamp = e.time().timeHigh(); // precision to the second 
+
+
+  // We work out the trigger information here 
+  if( !fTriggerLabel.empty() ) { 
+
+      art::Handle<std::vector<sim::BeamGateInfo>> beamgate_handle;
+      e.getByLabel( fTriggerLabel, beamgate_handle );
+
+      if( beamgate_handle.isValid() ) {
+
+        for( auto const & beamgate : *beamgate_handle ) {
+
+          m_beam_gate_start = beamgate.Start(); 
+          m_beam_gate_width = beamgate.Width(); 
+          m_beam_type = beamgate.BeamType() ;
+
+        }
+
+      }
+
+      else {
+        std::cout << "Invalid Trigger Data product " << fTriggerLabel.label() << "\n" ;
+      }
+
+  }
+
+  else {
+     std::cout << "Trigger Data product " << fTriggerLabel.label() << " not found!\n" ;
+  }
+
+
+  // Now we take care of the labels
+  if ( !fFlashLabels.empty() ) {
+
+    for ( size_t i=0; i<fFlashLabels.size(); i++  ) {
+
+      auto const label = fFlashLabels[i];
+
+      art::Handle<std::vector<recob::OpFlash>> flash_handle;
+      e.getByLabel( label, flash_handle );
+
+      if( flash_handle.isValid() ) {
```

Comment:
I have been told in the past that these modules should not be flexible if input data is not present. So this should crash if the handle is not valid.